managers/docker-manager/DockerManager.js

```diff
@@ -0,0 +1,95 @@
+'use strict';
+
+const Promise = require('bluebird');
+const eventmesh = require('../../data-access-layer/eventmesh');
+const logger = require('../../common/logger');
+const CONST = require('../../common/constants');
+const utils = require('../../common/utils');
+const BaseManager = require('../BaseManager');
+const DockerService = require('./DockerService');
+const errors = require('../../common/errors');
+const ServiceInstanceNotFound = errors.ServiceInstanceNotFound;
+
+class DockerManager extends BaseManager {
+
+  init() {
+    const validStateList = [CONST.APISERVER.RESOURCE_STATE.IN_QUEUE, CONST.APISERVER.RESOURCE_STATE.UPDATE, CONST.APISERVER.RESOURCE_STATE.DELETE];
+    return this.registerCrds(CONST.APISERVER.RESOURCE_GROUPS.DEPLOYMENT, CONST.APISERVER.RESOURCE_TYPES.DOCKER)
+      .then(() => this.registerWatcher(CONST.APISERVER.RESOURCE_GROUPS.DEPLOYMENT, CONST.APISERVER.RESOURCE_TYPES.DOCKER, validStateList));
+  }
+
+  processRequest(changeObjectBody) {
+    return Promise.try(() => {
+        if (changeObjectBody.status.state === CONST.APISERVER.RESOURCE_STATE.IN_QUEUE) {
+          return this._processCreate(changeObjectBody);
+        } else if (changeObjectBody.status.state === CONST.APISERVER.RESOURCE_STATE.UPDATE) {
+          return this._processUpdate(changeObjectBody);
+        } else if (changeObjectBody.status.state === CONST.APISERVER.RESOURCE_STATE.DELETE) {
+          return this._processDelete(changeObjectBody);
+        }
+      })
+      .catch(err => {
+        logger.error('Error occurred in processing request by DockerManager', err);
+        return eventmesh.apiServerClient.updateResource({
+          resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.DEPLOYMENT,
+          resourceType: CONST.APISERVER.RESOURCE_TYPES.DOCKER,
+          resourceId: changeObjectBody.metadata.name,
+          status: {
+            state: CONST.APISERVER.RESOURCE_STATE.FAILED,
+            error: utils.buildErrorJson(err)
+          }
+        });
+      });
+  }
+
+  _processCreate(changeObjectBody) {
+    const changedOptions = JSON.parse(changeObjectBody.spec.options);
+    logger.info('Creating docker resource with the following options:', changedOptions);
+    return DockerService.createDockerService(changeObjectBody.metadata.name, changedOptions)
```

Comment:
`DockerService.createDockerService` - does not read nice, instead can this be named `DockerService.createInstance` - generally followed naming convention for factory method.

Reply:
Added a TODO